src/Game/UI/Gumps/BookGump.cs

```diff
@@ -63,30 +64,38 @@ public BookGump(uint serial) : base(serial, 0)
         }
 
         public ushort BookPageCount { get; internal set; }
-        public static bool IsNewBookD4 => Client.Version > ClientVersion.CV_200;
-        public static byte DefaultFont => (byte) (IsNewBookD4 ? 1 : 4);
+        public static bool IsNewBook => Client.Version > ClientVersion.CV_200;
+        public bool UseNewHeader { get; set; } = true;
+        public static byte DefaultFont => (byte) (IsNewBook ? 1 : 4);
 
         public string[] BookPages
         {
-            get => null;
+            get
+            {
+                string[] pages = new string[BookPageCount];
+                for (int i = 0; i < pages.Length; i++) pages[i] = string.Empty;
+
+                for (int i = 0; i < m_Pages.Count; i++)
+                {
+                    pages[i] = m_Pages[i].Text;
+                }
+                return pages;
+            }
             set
             {
                 if (value != null)
                 {
                     if (_activated > 0)
                     {
-                        for (int i = 0; i < m_Pages.Count; i++)
+                        for (int i = 0; i < Math.Min(m_Pages.Count, value.Length); i++)
```

Comment:
Doing the math.min for each iteration it's a bit overheaded or is there a particular reason?